__tests__/createRetryFetch.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+const createRetryFetch = require('../src/createRetryFetch');
+
+describe('createRetryFetch', () => {
+  it('should resolve on success', async () => {
+    const mockFetch = jest.fn()
+      .mockImplementationOnce(() => Promise.resolve('Fetch Resolved'));
+    const enhancedFetch = createRetryFetch()(mockFetch);
+    await expect(enhancedFetch('/')).resolves.toBe('Fetch Resolved');
+  });
+
+  it('should resolve with invalid config', async () => {
+    const mockFetch = jest.fn()
+      .mockImplementationOnce(() => Promise.resolve('Fetch Resolved'));
+    const enhancedFetch = createRetryFetch({})(mockFetch);
+    await expect(enhancedFetch('/')).resolves.toBe('Fetch Resolved');
+  });
+
+  it('should reject on failure', async () => {
+    const mockFetch = jest.fn()
+      .mockImplementationOnce(() => Promise.reject(new Error('test rejection error')))
+      .mockImplementationOnce(() => Promise.reject(new Error('test rejection error')))
+      .mockImplementationOnce(() => Promise.reject(new Error('test rejection error')))
+      .mockImplementationOnce(() => Promise.reject(new Error('test rejection error')));
```

Comment:
Not requesting a change, but FWIW `jest.fn` can be used in neat ways to save duplication/effort:
```suggestion
    const mockFetch = jest.fn(() => Promise.reject(new Error('test rejection error')));
```
or
```suggestion
    const mockFetch = jest.fn()
      .mockImplementation(() => Promise.reject(new Error('test rejection error')));
```

Alternatively, if wanting to check which call's rejection propagated (though perhaps implementation-specific, so undesired in this instance?)
```suggestion
    const mockFetch = jest.fn()
      .mockImplementationOnce(() => Promise.reject(new Error('test rejection 1 error')))
      .mockImplementationOnce(() => Promise.reject(new Error('test rejection 2 error')))
      .mockImplementationOnce(() => Promise.reject(new Error('test rejection 3 error')))
      .mockImplementationOnce(() => Promise.reject(new Error('test rejection 4 error')));
```